spec/features/admin/timeline_batch_actions_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+feature 'timeline batch actions', js: true do
+  let(:division) { create(:division) }
+  let(:user) { create_member(division) }
+  let!(:loan) { create(:loan, :with_timeline, division: division) }
+
+  # Create a step with a dependent step, which also has a dependent step
+  let(:step1) { create(:project_step, project: loan) }
+  let(:step2) { create(:project_step, project: loan, scheduled_start_date: nil, schedule_parent: step1) }
+  let(:step3) { create(:project_step, project: loan, scheduled_start_date: nil, schedule_parent: step2) }
+
+  # Create two independent steps, one with and the other without a scheduled start date
+  let(:step4) { create(:project_step, project: loan, scheduled_start_date: Date.today, scheduled_duration_days: nil) }
+  let(:step5) { create(:project_step, project: loan, scheduled_start_date: nil, scheduled_duration_days: nil) }
+
+  before do
+    login_as(user, scope: :user)
+  end
+
+  it 'changes dates' do
+    # Add the steps to the loan
+    loan.root_timeline_entry.children += [step1, step2, step3, step4, step5]
+
+    # Set variables
+    step1_orig_date = step1.scheduled_start_date
+    step2_orig_date = step2.scheduled_start_date
+    step3_orig_date = step3.scheduled_start_date
+    step4_orig_date = step4.scheduled_start_date
+    step5_orig_date = step5.scheduled_start_date
+    num = 3
+
+    # Select steps
+    visit(admin_loan_path(loan) + '/timeline')
+    find(:css, ".select-step[data-id='#{step1.id}']").set(true)
+    find(:css, ".select-step[data-id='#{step2.id}']").set(true)
+    find(:css, ".select-step[data-id='#{step3.id}']").set(true)
+    find(:css, ".select-step[data-id='#{step4.id}']").set(true)
+    find(:css, ".select-step[data-id='#{step5.id}']").set(true)
```

Comment:
Might it make sense to _not_ select one of the schedule parents so that you can test that it gets appropriately skipped?